File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevMap = {}  # val -> index

        for i, n in enumerate(nums):
            diff = target - n
            if diff in prevMap: #checks for key
                return [prevMap[diff], i]
            prevMap[n] = i #updates the dict with elements and its respective index

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int a = 0;
        int b = numbers.length - 1;
        boolean found = false;
        int[] sum = new int[2];

        while (a <= b && found == false) {
            if (numbers[a] + numbers[b] == target) {
                sum[0] = a + 1;
                sum[1] = b + 1;
                found = true;
            }
            else if(numbers[a] + numbers[b] > target) {
                b -= 1;
            }
            else if(numbers[a] + numbers[b] < target) {
                a += 1;
            }
        }

        return sum;
    }
}
